assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,87 @@
-console.log('***** Music Collection *****')
+console.log('***** Music Collection *****');
+
+// global variabls
+let collection = [];
+
+// functions
+let addToCollection = (title, artist, yearPublished) => {
+    let album = {
+        title: title,
+        artist: artist,
+        yearPublished: yearPublished,
+    };
+    collection.push(album);
+    return album;
+};
+
+let showCollection = (albumArray) => {
+    console.log(`Number of albums: ${albumArray.length}`);
+    for (let album of albumArray) {
+        console.log(
+            `${album.title} by ${album.artist},` +
+                ` released in ${album.yearPublished}`
+        );
+    }
+};
+
+let findByArtist = (artist) => {
+    let artistCollection = [];
+    for (let album of collection) {
+        if (album.artist.toLowerCase() === artist.toLowerCase()) {
+            artistCollection.push(album);
+        }
+    }
+    return artistCollection;
+};
+
+let search = (criteria) => {
+    let matchingAlbums = [];
+    for (let album of collection) {
+        let isMatching = Object.entries(criteria).reduce(
+            (m, [key, value]) =>
+                Object.hasOwn(album, key) && album[key] === value && m,
+            true
+        );
+        if (isMatching) {
+            matchingAlbums.push(album);
+        }
+    }
+    return matchingAlbums;
+};
+
+// main
+addToCollection('Emotions and Math', 'Margaret Glaspy', 2016);
```

Comment:
The requirements for testing addToCollection specify that you must "Console.log each album **as added** using the returned value." Although you console.log the overall collection later, you never confirm that the return value for this function works as expected.